exp/views/responses.py

```diff
@@ -1109,6 +1165,78 @@ def get(self, request, *args, **kwargs):
         return response
 
 
+class StudyResponsesFrameDataPsychDS(ResponseDownloadMixin, generic.list.ListView):
+    """Hitting this URL downloads a ZIP file in Psych-DS formatting with frame data from one response per file in CSV format"""
+
+    # TODO: with large files / many responses generation can take a while. Should generate asynchronously along
+    # with the data dict.
+    def render_to_response(self, context, **response_kwargs):
+        paginator = context["paginator"]
+        study = self.study
+
+        if study.study_type.is_external:
+            messages.error(
+                self.request, "Frame data is not available for External Studies."
+            )
+
+            return study_responses_all(study)
+
+        # build dict with contextual infomation about the study/lab/user
+        metadata_json = build_metadata_object(study)
+        # variables from column headers will be collected while looping through files
+        variables_measured = []
+
+        zipped_file = io.BytesIO()
+        with zipfile.ZipFile(zipped_file, "w", zipfile.ZIP_DEFLATED) as zipped:
+            for page_num in paginator.page_range:
+                page_of_responses = paginator.page(page_num)
+                for resp in page_of_responses:
+                    data = build_single_response_framedata_csv(resp)
+                    # get column headers from first row
+                    variables_measured += [
+                        column.strip('"')
+                        for column in data.split("\n")[0].strip().split(",")
+                    ]
```

Comment:
When I tested this, my `variables_measured` array contained repeated values and variables called "key" and "value" (see below). Just wondering if that's what you'd expect?
```json
 "variableMeasured": [
        "response_uuid",
        "child_hashed_id",
        "frame_id",
        "event_number",
        "key",
        "value",
        "response_uuid",
        "child_hashed_id",
        "frame_id",
        "event_number",
        "key",
        "value",
        "response_uuid",
        "child_hashed_id",
        "frame_id",
        "event_number",
        "key",
        "value",
        "response_uuid",
        "child_hashed_id",
        "frame_id",
        "event_number",
        "key",
        "value",
        "response_uuid",
        "child_hashed_id",
        "frame_id",
        "event_number",
        "key",
        "value"
    ]
```